Mathlib/Topology/Order/ScottTopology.lean

```diff
@@ -325,11 +325,9 @@ lemma continuous_iff_scottContinuous
       have e1: IsLUB (f '' d) (f a) := h hd₁ hd₂ hd₃
       rw [isOpen_iff_upper_and_LUB_mem_implies_inter_nonempty] at hu
       have e2: ((f '' d) ∩ u).Nonempty := by
-        apply hu.2 _ _ (Nonempty.image f hd₁)
-        apply (directedOn_image.mpr (DirectedOn.mono hd₂ _))
+        apply hu.2 (f '' d) (f a) (Nonempty.image f hd₁)
+          (directedOn_image.mpr (DirectedOn.mono hd₂ _)) e1 ha
```

Comment:
It's good style to leave arguments that can be unified as placeholders. Also there's dot notation available.
```suggestion
        apply hu.2 _ _(hd₁.image f ) (directedOn_image.mpr (hd₂.mono _)) e1 ha
```

Reply:
@YaelDillies Thanks, I've made this change:
```
      have e2: ((f '' d) ∩ u).Nonempty := by
        apply hu.2 _ _ (hd₁.image f)
          (directedOn_image.mpr (hd₂.mono _)) (h hd₁ hd₂ hd₃) ha
        apply h.monotone
```
It feels like I should be able to merge these two `apply` statements into a single line - but I can't figure out how to do this?

Reply:
I would say rather "it is good style to leave arguments which are not helpful to a human reader, and can by unified, as placeholders". :-)

Reply:
Finger-guessing
```
      have e2 : ((f '' d) ∩ u).Nonempty :=
        hu.2 _ _ _ (directedOn_image.mpr $ hd₂.mono h.monotone) (h hd₁ hd₂ hd₃) ha
```
You could even get down to
```
      have e2 : ((f '' d) ∩ u).Nonempty :=
        hu.2 _ _ _ (hd₂.mono h.monotone).image (h hd₁ hd₂ hd₃) ha
```
if you added
```
alias directedOn_image ↔ DirectedOn.of_image DirectedOn.image
```
in `order.directed`

Reply:
> ```
>       have e2 : ((f '' d) ∩ u).Nonempty :=
>         hu.2 _ _ _ (directedOn_image.mpr $ hd₂.mono h.monotone) (h hd₁ hd₂ hd₃) ha
> ```
@YaelDillies That's what I was expecting, but it doesn't work:
```
type mismatch
  ScottContinuous.monotone h
has type
  Monotone f : Prop
but is expected to have type
  a✝ ≤ b✝ → (f ⁻¹'o fun x x_1 ↦ x ≤ x_1) a✝ b✝ : Prop
```
The closest I could come up with was:
```
      have e2: ((f '' d) ∩ u).Nonempty := hu.2 _ _ (hd₁.image f)
          (directedOn_image.mpr (hd₂.mono (by simp only [Order.Preimage]; apply h.monotone)))
          (h hd₁ hd₂ hd₃) ha
```